opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1517,8 +1521,20 @@ namespace Opm {
         updatePrimaryVariables(deferred_logger);
     }
 
+    template<typename TypeTag>
+    void
+    BlackoilWellModel<TypeTag>::
+    setupCartesianToCompressed_()
+    {
+        const auto& cartMapper = ebosSimulator_.vanguard().cartesianIndexMapper();
+        const size_t cartesianSize = cartMapper.cartesianSize();
+        cartesian_to_compressed_.resize(cartesianSize, -1);
 
-
+        for (unsigned i = 0; i < local_num_cells_; ++i) {
+            unsigned cartesianCellIdx = cartMapper.cartesianIndex(i);
+            cartesian_to_compressed_[cartesianCellIdx] = i;
+        }
+    }
```

Comment:
Do we maybe have access ti an existing one (ebos/eclbasevanguard.hh, ebos/eclproblem.hh) here and can skip this?